src/http/server_context.rs

```diff
@@ -123,6 +65,94 @@ impl ServerContext {
       cache: ChronoCache::new(),
       grpc_data_loaders: Arc::new(grpc_data_loaders),
       env_vars,
+    };
+
+    for def in bp.definitions.iter_mut() {
+      if let Definition::ObjectTypeDefinition(def) = def {
+        for field in &mut def.fields {
+          if let Some(expr) = field.resolver.clone() {
+            ctx.check_resolver_of_field(field, &expr);
+          }
+        }
+      }
+    }
+    ctx.schema = bp.to_schema();
+    ctx.blueprint = bp;
+    ctx
+  }
+
+  fn check_resolver_of_field(&mut self, field: &mut blueprint::FieldDefinition, expr: &Expression) {
+    match expr {
+      Expression::Unsafe(expr_unsafe) => self.check_unsafe_expr(expr_unsafe, field),
+      Expression::Cache(cache) => {
+        self.check_resolver_of_field(field, cache.source());
+        field.resolver = field.resolver.as_ref().map(|ne| {
+          Expression::Cache(Cache::new(
+            cache.hasher().clone(),
+            cache.max_age(),
+            Box::new(ne.clone()),
+          ))
+        });
```

Comment:
do we need to do this? it look like a clone for cache expression without any changes

Reply:
check this again, here is definitely something wrong. The `ne` here is already a Cache instance leading to wrapping cache inside cache. You can check that by putting debug logs here or inside the expression's eval for cache